packages/syft/src/syft/serde/json_serde.py

```diff
@@ -0,0 +1,457 @@
+    """
+    recursively unwrap type annotations, removing Annotated and Optional types
+    """
+    if _is_annotated_type(annotation):
+        res = _unwrap_annotated(annotation)
+        return _unwrap_type_annotation(res)
+    elif _is_optional_annotation(annotation):
+        res = _unwrap_optional_annotation(annotation)
+        return _unwrap_type_annotation(res)
+    return annotation
+
+
+def _annotation_issubclass(annotation: Any, cls: type) -> bool:
+    # issubclass throws TypeError if annotation is not a valid type (eg Union)
+    try:
+        return issubclass(annotation, cls)
+    except TypeError:
+        return False
+
+
+def _serialize_pydantic_to_json(obj: pydantic.BaseModel) -> dict[str, Json]:
+    canonical_name, version = SyftObjectRegistry.get_canonical_name_version(obj)
+    serde_attributes = SyftObjectRegistry.get_serde_properties(canonical_name, version)
+    exclude_attrs = serde_attributes[4]
+
+    result: dict[str, Json] = {
+        JSON_CANONICAL_NAME_FIELD: canonical_name,
+        JSON_VERSION_FIELD: version,
+    }
+
+    all_exclude_attrs = set(exclude_attrs) | DEFAULT_EXCLUDE_ATTRS
+
+    for key, type_ in obj.model_fields.items():
+        if key in all_exclude_attrs:
+            continue
+        result[key] = serialize_json(getattr(obj, key), type_.annotation)
+
+    result = _serialize_searchable_attrs(obj, result, raise_errors=False)
+
+    return result
+
+
+def get_property_return_type(obj: Any, attr_name: str) -> Any:
+    """
+    Get the return type annotation of a @property.
+    """
+    cls = type(obj)
+    attr = getattr(cls, attr_name, None)
+
+    if isinstance(attr, property):
+        return attr.fget.__annotations__.get("return", None)
+
+    return None
+
+
+def _serialize_searchable_attrs(
+    obj: pydantic.BaseModel, obj_dict: dict[str, Json], raise_errors: bool = True
+) -> dict[str, Json]:
+    """
+    Add searchable attrs  and unique attrs to the serialized object dict, if they are not already present.
+    Needed for adding non-field attributes (like @property)
+
+    Args:
+        obj (pydantic.BaseModel): Object to serialize.
+        obj_dict (dict[str, Json]): Serialized object dict. Should contain the object's fields.
+        raise_errors (bool, optional): Raise errors if an attribute cannot be accessed.
+            If False, the attribute will be skipped. Defaults to True.
+
+    Raises:
+        Exception: Any exception raised when accessing an attribute.
+
+    Returns:
+        dict[str, Json]: Serialized object dict including searchable attributes.
+    """
+    searchable_attrs: list[str] = getattr(obj, "__attr_searchable__", [])
+    unique_attrs: list[str] = getattr(obj, "__attr_unique__", [])
+
+    attrs_to_add = set(searchable_attrs) | set(unique_attrs)
+    for attr in attrs_to_add:
+        if attr not in obj_dict:
+            try:
+                value = getattr(obj, attr)
+            except Exception as e:
+                if raise_errors:
```

Comment:
when would you not want to do this?

Reply:
searchable attrs can be @property methods, and those can throw errors. Example: `Job.code_id` was throwing errors when the Job was for a twinAPI. This flag is to control this behaviour, 'do i want to skip writing if some method throws an error'